lib/reporters/tap.js

```diff
@@ -77,3 +79,95 @@ function TAP(runner) {
 function title(test) {
   return test.fullTitle().replace(/#/g, '');
 }
+
+/**
+ * Inherit from `Base.prototype`.
+ */
+inherits(TAP, Base);
+
+/**
+ * Initialize a new `TAPProducer`. Should only be used as a base class.
+ *
+ * @private
+ * @class
+ * @api private
+ */
+function TAPProducer() {}
+
```

Comment:
`TAP12Producer.prototype.writeStart` should be reparented as `TAPProducer.prototype.writeStart`.

Reply:
`TAPProducer.prototype.writeStart = function(ntests) {`